src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,208 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int listCount;
+
+    class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return String.valueOf(value);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(value);
+        if (first == null) {
+            newNode.next = null;
+            newNode.prev = null;
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        listCount++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
+        Node<T> newNode = new Node(value);
+        if (index == 0 && listCount == 0) {
+            add(value);
+        } else if (index == 0 && listCount > 0) {
+            newNode.next = first;
+            newNode.prev = null;
+            first.prev = newNode;
+            first = newNode;
+            listCount++;
+        } else if (index == listCount) {
+            add(value);
+        } else {
+            Node<T> oldNode = first;
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            addItem(oldNode, newNode);
+            listCount++;
+
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= listCount) {
+            throw new IndexOutOfBoundsException("index is wrong");
+        }
```

Comment:
you have the same logic in the method below, let's create a separate method `checkIndex`